al_servicec.js

```diff
@@ -143,25 +143,29 @@ class AlServiceC extends m_alUtil.RestServiceClient {
  * @param {Object} aisCreds - Alert Logic API credentials.
  * @param {string} [aisCreds.access_key_id] - Alert Logic API access key id.
  * @param {string} [aisCreds.secret_key] - Alert Logic API secret key.
+ * @param {string} contentType - Content-Type for Ingest.
+ * @param {string} ingestType - Ingest endpoint type.
  *
  */
 class IngestC extends AlServiceC {
-    constructor(apiEndpoint, aimsCreds) {
+    constructor(apiEndpoint, aimsCreds, contentType, ingestType) {
```

Comment:
Just a thought, but rather than allowing freeform types, wouldn't it be better to restrict types to an atomic list?  Valid types would be "cwe" etc. and then map to a final Content-Type string of `'alertlogic.com/cwe-json'` for example.

Reply:
I would prefer to keep it as it is. It allows us to modify only a collector without touching the library with every new type.

Reply:
I agree with @ikemsley. I don't like the idea of having open ingest types and would prefer to have them predefined. I see no problem with adding a new function as new API endpoint is added into into Ingest. Let's just have it incapsulated in this class.

Reply:
BTW, if we have such an open constructor I don't see any reason for having `Ingestc` class.
Also a new parameter (`x-invoked-by`) will be required if we add Azure function support.